scripts/globals/baitandswitch.lua

```diff
@@ -0,0 +1,203 @@
+    return ProgressCheck;
+end;
+
+function CheckSnares(player)
+    local kilnSnare = 0;
+    local shopSnare = 0;
+    local refinerySnare = 0;
+    local replicaSnare = 0;
+    local gunpowderSnare = 0;
+    local cidSnare = 0;
+    local snareNumber = 0;
+
+    if (player:getMaskBit(player:getLocalVar("Bait_and_Switch_Quest_Option"),9) == true) then
+        kilnSnare = 1;
+    end
+    if (player:getMaskBit(player:getLocalVar("Bait_and_Switch_Quest_Option"),10) == true) then
+        shopSnare = 1;
+    end
+    if (player:getMaskBit(player:getLocalVar("Bait_and_Switch_Quest_Option"),11) == true) then
+        refinerySnare = 1;
+    end
+    if (player:getMaskBit(player:getLocalVar("Bait_and_Switch_Quest_Option"),12) == true) then
+        replicaSnare = 1;
+    end
+    if (player:getMaskBit(player:getLocalVar("Bait_and_Switch_Quest_Option"),13) == true) then
+        gunpowderSnare = 1;
+    end
+    if (player:getMaskBit(player:getLocalVar("Bait_and_Switch_Quest_Option"),14) == true) then
+        cidSnare = 1;
+    end
+
+    local SnareCheck = {kilnSnare,shopSnare,refinerySnare,replicaSnare,gunpowderSnare,cidSnare};
+    for i,v in ipairs(SnareCheck) do
+        if (v == 1) then
+            snareNumber = snareNumber + 1;
+        end
+    end
+    local SnareCheckfin = {kilnSnare,shopSnare,refinerySnare,replicaSnare,gunpowderSnare,cidSnare,snareNumber};
+    return SnareCheckfin;
+end;
+
+function RecordBaitProgress(player)
+    local BitConvert = {0,1,2,3,4,5,6};
+    local ChosenItem = CheckOptionForItem(player);
+    local Progress = player:getVar("Bait_and_Switch_Quest_Progress");
+    player:setMaskBit(Progress,"Bait_and_Switch_Quest_Progress",BitConvert[ChosenItem],true);
+end;
+
+function CheckOptionForItem(player)
+    local OptionCheck = 0;
+    if (player:getLocalVar("Bait_and_Switch_Quest_Option") == 1) then
+        OptionCheck = 1;
+    elseif (player:getLocalVar("Bait_and_Switch_Quest_Option") == 2) then
+        OptionCheck = 2;
+    elseif (player:getLocalVar("Bait_and_Switch_Quest_Option") == 3) then
+        OptionCheck = 3;
+    elseif (player:getLocalVar("Bait_and_Switch_Quest_Option") == 0) then
+        OptionCheck = 4;
+    elseif (player:getLocalVar("Bait_and_Switch_Quest_Option") == 4 or
+            player:getLocalVar("Bait_and_Switch_Quest_Option") == 5 or
+            player:getLocalVar("Bait_and_Switch_Quest_Option") == 6) then
+        OptionCheck = 5;
+    elseif (player:getLocalVar("Bait_and_Switch_Quest_Option") == 7) then
+        OptionCheck = 6;
+    elseif (player:getLocalVar("Bait_and_Switch_Quest_Option") == 8) then
+        OptionCheck = 7;
+    elseif (player:getLocalVar("Bait_and_Switch_Quest_Option") >= 512) then
+        OptionCheck = 4;
+    end
+    return OptionCheck;
+end;
+
+local S = 1;
+local M = 2;
+local L = 3;
+
+Switch_Table =
+{
+-- Scope
+    [101] = {ord = {S,M,L},snake = 3,helmut = 0,wolf = 2,darha = 1,digits = 3},
```

Comment:
```lua
digits = val
```
can be omitted
instead, wherever you'd check `digits` you can just check
```lua
local totalDigits = #switch_table[key].ord
```

Reply:
yep, that makes sense... im on it

Reply:
fixed